vendor/bat-native-ads/src/bat/ads/internal/ml/data/vector_data.cc

```diff
@@ -137,6 +138,61 @@ double operator*(const VectorData& lhs, const VectorData& rhs) {
   return dot_product;
 }
 
+void VectorData::VectorAddElementWise(const VectorData& v_add) {
+  if (!storage_->dimension_count() || !v_add.storage_->dimension_count()) {
+    return;
+  }
+
+  if (storage_->dimension_count() != v_add.storage_->dimension_count()) {
+    return;
+  }
+
+  size_t v_base_index = 0;
+  size_t v_add_index = 0;
+  while (v_base_index < storage_->GetSize() &&
+         v_add_index < v_add.storage_->GetSize()) {
+    if (storage_->GetPointAt(v_base_index) ==
+        v_add.storage_->GetPointAt(v_add_index)) {
+      storage_->values()[v_base_index] += v_add.storage_->values()[v_add_index];
+      ++v_base_index;
+      ++v_add_index;
+    } else {
+      if (storage_->GetPointAt(v_base_index) <
+          v_add.storage_->GetPointAt(v_add_index)) {
+        ++v_base_index;
+      } else {
+        ++v_add_index;
+      }
+    }
+  }
+}
+
+void VectorData::VectorDivideByScalar(float scalar) {
+  if (!storage_->dimension_count()) {
+    return;
+  }
+
+  size_t v_index = 0;
+  while (v_index < storage_->GetSize()) {
+    storage_->values()[v_index] /= scalar;
+    ++v_index;
+  }
+}
+
+float VectorData::VectorSumElements() {
```

Comment:
we might not need this function anymore, but I'm fine with keeping it as it may probably turn out useful in the future

Reply:
We should remove unused code

Reply:
Removed